### DISCUSSION TITLE
Good way to extract small parts of YouTube videos as you watch

### DISCUSSION THREAD

**Discussion by andrewarrow** [FreeTubeApp/FreeTube#5235] [Q&A] [answered]
I want right in the gui where the video is playing being able to mark timecode1,timecode2 and then download just that small clip.

Flow like this: hit spacebar to pause, then press A to mark that SpotA and then hit space bar again to resume playing... space again to pause, then press B to mark that SpotB. Download and repeat again for another section in video.

Before I build myself a gui to do this, is there a way in freetube?

**Comment by absidue** [comment, marked as answer]
Something like that is not supported in FreeTube and will likely never be, as it's a very niche use case that would require a lot of very complex work to implement inside the app.

I would recommend watching the video in FreeTube, writing down all the time ranges and then using `yt-dlp` with it's `--download-sections` option.

https://github.com/yt-dlp/yt-dlp

https://github.com/yt-dlp/yt-dlp/blob/master/README.md#user-content-download-options
